internal/cri/server/events.go

```diff
@@ -379,6 +379,12 @@ func (em *eventMonitor) handleEvent(any interface{}) error {
 
 // handleContainerExit handles TaskExit event for container.
 func handleContainerExit(ctx context.Context, e *eventtypes.TaskExit, cntr containerstore.Container, sandboxID string, c *criService) error {
+	// First we need to update the status of container to Exiting
+	// to avoid container updating from UpdateContainerResources
+	_ = cntr.Status.Update(func(status containerstore.Status) (containerstore.Status, error) {
+		status.Exiting = true
```

Comment:
The change looks good. But I think we can consider to remove the lock for UpdateContainerResource.

ping @mikebrow for more input about why we should lock store before updating container resource.
Even if there is data race, kubelet can reconcile it to correct state .

Reply:
https://github.com/containerd/containerd/blob/79a3b20a63e86f8cfb1f23dda99ca26bd576daf9/internal/cri/server/container_update_resources.go#L60C1-L67C3

Reply:
Good idea.

`UpdateContainerResource` is blocked by `task.Update`, but `Status.UpdateSync` don't depends result from `task.Update`. The lock maybe is necessary, but we can reduce the critical section of the lock.




https://github.com/containerd/containerd/blob/79a3b20a63e86f8cfb1f23dda99ca26bd576daf9/internal/cri/server/container_update_resources.go#L60C1-L67C3

Reply:
Hi @mikebrow, is there any update of this?